scaleway/resource_storage_object_bucket.go

```diff
@@ -0,0 +1,113 @@
+package scaleway
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/scaleway/scaleway-sdk-go/utils"
+)
+
+func resourceScalewayStorageObjectBucket() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceScalewayStorageObjectBucketCreate,
+		Read:   resourceScalewayStorageObjectBucketRead,
+		Update: resourceScalewayStorageObjectBucketUpdate,
+		Delete: resourceScalewayStorageObjectBucketDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The name of the bucket",
+			},
+			"acl": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "private",
+				Description: "ACL of the bucket: either 'public-read' or 'private'.",
+			},
+			"region": regionSchema(),
+		},
+	}
+}
+
+func resourceScalewayStorageObjectBucketCreate(d *schema.ResourceData, m interface{}) error {
+	bucketName := d.Get("name").(string)
+	acl := d.Get("acl").(string)
+
+	s3Client, region, err := getS3ClientWithRegion(d, m)
+
+	_, err = s3Client.CreateBucket(&s3.CreateBucketInput{
+		Bucket: aws.String(bucketName),
+		ACL:    aws.String(acl),
+	})
+	if err != nil {
+		return err
+	}
+
+	d.SetId(newRegionalId(utils.Region(region), bucketName))
+
+	return resourceScalewayStorageObjectBucketRead(d, m)
+}
+
+func resourceScalewayStorageObjectBucketRead(d *schema.ResourceData, m interface{}) error {
+	s3Client, _, bucketName, err := getS3ClientWithRegionAndID(d, m)
```

Comment:
please use same interface os `getInstanceapiWithRegionAndID` (take and id not d)
Usefull for tests